src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Abstractions/Mappers/AssemblyMapper.cs

```diff
@@ -15,15 +13,26 @@ namespace Microsoft.DotNet.ApiCompatibility.Abstractions
     /// </summary>
     public class AssemblyMapper : ElementMapper<IAssemblySymbol>
     {
-        private Dictionary<INamespaceSymbol, NamespaceMapper> _namespaces;
+        private Dictionary<INamespaceSymbol, NamespaceMapper>? _namespaces;
+        private readonly List<CompatDifference>[] _assemblyLoadErrors;
+
+        /// <summary>
+        /// Gets the assembly load errors that happened when trying to follow type forwards.
+        /// </summary>
+        public IReadOnlyList<IReadOnlyList<CompatDifference>> AssemblyLoadErrors => _assemblyLoadErrors;
```

Comment:
Moving the AssemblyLoadErrors from the the base ElementMapper<T> type into the AssemblyMapper as that's the only mapper that deals with resolving external assemblies (via type forwards). This reduces the instantiation costs of the other mapper types.

Reply:
That seems fine by me. One small comment is that I'm not sure I understand why is that relevant in the nullability changes.

Reply:
Yes, this isn't related to the nullable reference type feature. This is general code clean-up (same as for the other comments that I left on this PR) that I bundled into this PR as I already touched the files.